src/track/beatutils.cpp

```diff
@@ -13,49 +13,181 @@
 
 #include "track/beatutils.h"
 #include "util/math.h"
+#include "track/beatstats.h"
 
-// we are generous and assume the global_BPM to be at most 0.05 BPM far away
-// from the correct one
-#define BPM_ERROR 0.05
+// we are generous and assume the global_BPM to be at most 0.1 BPM far away
+#define BPM_ERROR 0.1 // changing this from 0.05 to 0.1 yields better results
 
 // the raw beatgrid is divided into blocks of size N from which the local bpm is
 // computed. Tweaked from 8 to 12 which improves the BPM accuracy for 'problem songs'.
 #define N 12
 
-static bool sDebug = false;
+constexpr int kBeatsToCountTempo = 12;
+
+static bool sDebug = true;
 
 const double kCorrectBeatLocalBpmEpsilon = 0.05; //0.2;
 const int kHistogramDecimalPlaces = 2;
 const double kHistogramDecimalScale = pow(10.0, kHistogramDecimalPlaces);
 const double kBpmFilterTolerance = 1.0;
 
-void BeatUtils::printBeatStatistics(const QVector<double>& beats, int SampleRate) {
-    if (!sDebug) {
-        return;
+double BeatUtils::medianBpm(const QVector<double>& beats, int SampleRate,
+                               int min_bpm, int max_bpm) {
+
+    if (beats.size() < 2) {
+        return 0;
+    }
+
+    // If we don't have enough beats for our regular approach, just divide the #
+    // of beats by the duration in minutes.
+    if (beats.size() <= 2) {
+        return 60.0 * (beats.size()-1) * SampleRate / (beats.last() - beats.first());
```

Comment:
```suggestion
        return 60.0 * (beats.size()-1) * sampleRate / (beats.last() - beats.first());
```
Capitalizing the name of a function parameter breaks our conventions. I was confused at first because it looked like a type name.